src/page.rs

```diff
@@ -276,19 +279,25 @@ impl PageReader {
             return Ok(0);
         }
 
-        if self.chunk_pos == self.ch.chunk_len {
-            trace!("read: at end of chunk");
-            if !self.next_chunk(flash).await? {
-                trace!("read: no next chunk, we're at end.");
-                return Ok(0);
+        let mut read = 0;
+        loop {
```

Comment:
I wouldn't add a loop here. the `file` layer already loops, this will increase code size a bit for no benefit I think.

(if you did it for tests, i'd either update the tests or add a helper just for tests that does loop)